src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php

```diff
@@ -185,47 +185,84 @@ public function editAction(Request $request, $id)
 
                 return new RedirectResponse($this->generateUrl(
                     $indexUrl['path'],
-                    isset($indexUrl['params']) ? $indexUrl['params'] : array()
+                    isset($indexUrl['params']) ? $indexUrl['params'] : []
                 ));
             }
         }
 
-        return array(
+        return [
             'form' => $form->createView(),
             'translation' => $translation,
-            'adminlistconfigurator' => $configurator
-        );
+            'adminlistconfigurator' => $configurator,
+        ];
+    }
+
+    /**
+     * @Route("upload", name="KunstmaanTranslatorBundle_settings_translations_upload")
+     * @Method({"POST", "GET"})
+     * @Template("KunstmaanTranslatorBundle:Translator:addTranslation.html.twig")
+     *
+     * @param Request $request
+     * @return array
+     */
+    public function uploadFileAction(Request $request)
+    {
+        $data = [];
+        /** @var FormBuilderInterface $uploadForm */
+        $form = $this->createFormBuilder($data)->add('file', FileType::class, [])->getForm();
+        $configurator = $this->getAdminListConfigurator();
+
+        if ('POST' === $request->getMethod()) {
+            $form->handleRequest($request);
+            if ($form->isSubmitted() && $form->isValid()) {
+                $locales = explode('|', $this->getParameter('requiredlocales'));
+                $data = $form->getData();
+                $file = $data['file'];
+                $imported = $this->get('kunstmaan_translator.service.importer.importer')->importFromSpreadsheet($file, $locales, false);
```

Comment:
I think we need a extra checkbox on the upload form if you want to force import translations. I can imagine you want to import translations and overwrite existing ones ?